### PR TITLE
[Move] Implement True Force Switch (Roar, Whirlwind, Circle Throw, Dragon Tail)

### LINKED ISSUE #4296: [Bug] True force-switch (Roar, Whirlwind, Circle Throw, Dragon Tail...)
## Describe the bug

Right now the player can choose their Pokemon which makes these moves actually beneficial for the player, instead of "punishing" him

## Expected behavior

These moves should do a proper force-switch meaning choosing a Pokemon at random.
This would also match the behavior from the original games.

## Additional context

There was a PR https://github.com/pagefaultgames/pokerogue/pull/1927 but unfortunately it was abandoned and is now too out of date to adopt it.
<sub>The original author has no intention to update it either</sub>

### PR DESCRIPTION
**Pull request by EvasiveAce**
## What are the changes the user will see?
Force switching moves, such as Roar, will now randomly switch to a party member if applicable. 

## Why am I making these changes?
Fixes #4296 
Turns the partial moves into fully implemented.

## What are the changes from a developer perspective?
`SwitchType.FORCE_SWITCH` added to `SwitchType` enum, logic to handle it added to `ForceSwitchOutAttr`.

### Screenshots/Videos

<details>
  <summary>Battle to show that Trainers are random; the previous version would not allow combusken to be sent out if selected, due to the type advantage, as far as I know.</summary>

[Video](https://github.com/user-attachments/assets/708d1268-d807-478f-814a-3e409d478f76)

</details>
<details>
  <summary>Battle to show that Roar works on the player.</summary>

[Video](https://github.com/user-attachments/assets/118dd132-3da4-4a93-bb0f-bbed56a5a48c)

</details>

## How to test the changes?
Make the trainer have all roar while you have any non damaging move. Vice verse to test the other side.

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I considered writing automated tests for the issue?
- [x] If I have text, did I make it translatable and add a key in the English locale file(s)?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
    
image

- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Review by DayKev — COMMENTED** [review]
Noticed a couple mistakes, PR here: https://github.com/EvasiveAce/pokerogue/pull/1

File: src/data/move.ts [hunk 1]
```diff
@@ -5979,7 +5979,21 @@ export class ForceSwitchOutAttr extends MoveEffectAttr {
         return false;
       }
 
-      if (switchOutTarget.hp > 0) {
+      if (switchOutTarget.hp && SwitchType.FORCE_SWITCH) {
```
Comment (DayKev) [inline src/data/move.ts:5982]: Here, the `this.switchType ===` is missing.

File: src/data/move.ts [hunk 2]
```diff
@@ -5979,7 +5979,21 @@ export class ForceSwitchOutAttr extends MoveEffectAttr {
...
+        switchOutTarget.leaveField(this.switchType === SwitchType.FORCE_SWITCH);
+        const slotIndex = Utils.randIntRange(user.scene.currentBattle.getBattlerCount(), user.scene.getPlayerParty().length);
+        user.scene.unshiftPhase(
```
Comment (DayKev) [inline src/data/move.ts:5985]: The existing code uses `prependToPhase(..., MoveEndPhase)` so I swapped the new code to use that as well.

**Review by DayKev — COMMENTED** [review]
Probably should mention exactly what I changed, so:

**Review by DayKev — APPROVED** [review]

**Comment by DayKev** [comment]
Fixed some comment placement I forgot about when making the review PR. Should be good to go now I think.

**Review by innerthunder — APPROVED** [review]

**Pull request merged**